src/components/Authentication/LoginRedirect.js

```diff
@@ -0,0 +1,46 @@
+import React, { Component } from 'react'
+import { Auth } from "aws-amplify";
+import { setUser } from '../../actions/UserActions'
+import { connect } from "react-redux";
+import { Redirect } from "react-router-dom";
+import CircularProgress from '@material-ui/core/CircularProgress';
+
+export class LoginRedirect extends Component {
+
+    getAuthenticatedUser() {
+        Auth.currentAuthenticatedUser()
+            .then(user => {
+                const email = user.attributes.email
+                if (email.substring(email.indexOf("@") + 1, email.length) === 'ubcbiztech.com') {
+                    this.props.setUser(user)
+                }
+                else {
+                    Auth.signOut()
+                    alert('You must use a ubcbiztech.com email')
+                }
+            })
+            .catch(() => console.log("Not signed in"))
+    }
+
+    componentDidMount() {
+        // Check if there is an authenticated user
+        // After authenticating with Google, Auth redirects to localhost:3000/login/ 
+        // There is a time delay where Auth.currentAuthenticatedUser() returns the wrong value. Hence setTimeout()
+        setTimeout(() => this.getAuthenticatedUser(), 800)
```

Comment:
Will test this feature later tonight

So to my understanding, the app now waits 800ms to allow for proper authentication to be completed; however, if authentication returns the wrong value and persists with the wrong value, then we ask the user to refresh?

As well, is the 800ms still just an arbitrary choice

Reply:
Pretty much. Do you think we should use polling instead?

Reply:
To be precise, it shows "If this is a taking a long time" and the progress bar until the user is populated. And if the authentication returns the wrong value, the user is never populated.

Reply:
> Pretty much. Do you think we should use polling instead?

should be okay, unless we see problems in the future